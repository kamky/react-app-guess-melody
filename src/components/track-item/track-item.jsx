// React
import React from "react";
import PropTypes from "prop-types";

// Component TrackItem
export const TrackItem = (props) => {
  const {index, userAnswer, onInputChange, renderAudioPlayer, answer} = props;

  const handleChange = (evt) => {
    const value = evt.target.checked;
    onInputChange(index, value);
  };

  return (
    <div className="track">
      {renderAudioPlayer(answer.src, index)}
      <div className="game__answer">
        <input
          className="game__input visually-hidden"
          type="checkbox"
          name="answer"
          value={`answer-${index}`}
          id={`answer-${index}`}
          checked={userAnswer}
          onChange={handleChange}
        />
        <label className="game__check" htmlFor={`answer-${index}`}>
          Отметить
        </label>
      </div>
    </div>
  );
};

TrackItem.propTypes = {
  index: PropTypes.number.isRequired,
  answer: PropTypes.shape({
    src: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
  }).isRequired,
  userAnswer: PropTypes.bool.isRequired,
  onInputChange: PropTypes.func.isRequired,
  renderAudioPlayer: PropTypes.func.isRequired,
};
