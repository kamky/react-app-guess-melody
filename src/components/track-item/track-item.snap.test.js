// React
import React from "react";
import renderer from "react-test-renderer";

// Components
import {TrackItem} from "./track-item";

describe(`Snapshot test for component <TrackItem />`, () => {
  const defaultProps = {index: 0, userAnswer: false, onInputChange: jest.fn()};
  const wrapper = renderer.create(<TrackItem {...defaultProps} />);
  test(`should component <TrackItem /> renders correctly`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
