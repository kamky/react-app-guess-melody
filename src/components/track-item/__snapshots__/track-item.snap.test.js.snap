// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for component <TrackItem /> should component <TrackItem /> renders correctly 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
  <div
    className="game__answer"
  >
    <input
      checked={false}
      className="game__input visually-hidden"
      id="answer-0"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-0"
    />
    <label
      className="game__check"
      htmlFor="answer-0"
    >
      Отметить
    </label>
  </div>
</div>
`;
